examples/apps/branching/src/container/runtimeFactory.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { getDataStoreEntryPoint } from "@fluid-example/example-utils";
+import type {
+	IContainerContext,
+	IRuntime,
+	IRuntimeFactory,
+} from "@fluidframework/container-definitions/legacy";
+import { loadContainerRuntime } from "@fluidframework/container-runtime/legacy";
+import type { IContainerRuntime } from "@fluidframework/container-runtime-definitions/legacy";
+import type { FluidObject } from "@fluidframework/core-interfaces";
+
+import { getChangesFromHealthBot } from "./getChangesFromHealthBot.js";
+import {
+	applyDiffToGroceryList,
+	GroceryListFactory,
+	type GroceryListChanges,
+	type IGroceryList,
+} from "./groceryList/index.js";
+
+const groceryListId = "grocery-list";
+const groceryListRegistryKey = "grocery-list";
+const groceryListFactory = new GroceryListFactory();
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type PrivateChanges = {
+	/**
+	 * A catalog of the changes that will be applied if acceptChanges is called.
+	 */
+	readonly changes: GroceryListChanges;
+	/**
+	 * Apply the changes to the grocery list.
+	 */
+	readonly acceptChanges: () => void;
+	/**
+	 * Drop the changes and return to the original grocery list.
+	 */
+	readonly rejectChanges: () => void;
+};
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type GroceryListAppEntryPoint = {
+	readonly groceryList: IGroceryList;
+	/**
+	 * Requests a service to provide suggested changes in a private state (not actually applied to
+	 * the collaborative data yet).  The PrivateChanges can be reviewed, and then accepted or rejected.
+	 */
+	readonly getSuggestions: () => Promise<PrivateChanges>;
+};
+
+export class GroceryListContainerRuntimeFactory implements IRuntimeFactory {
+	public get IRuntimeFactory(): IRuntimeFactory {
+		return this;
+	}
+
+	public async instantiateRuntime(
+		context: IContainerContext,
+		existing: boolean,
+	): Promise<IRuntime> {
+		const provideEntryPoint = async (
+			containerRuntime: IContainerRuntime,
+		): Promise<FluidObject> => {
+			const groceryList = await getDataStoreEntryPoint<IGroceryList>(
+				containerRuntime,
+				groceryListId,
+			);
+			const getSuggestions = async () => {
+				const changes = await getChangesFromHealthBot(groceryList);
+				// TODO: Here try integrating staging mode and applying the changes in that mode.
```

Comment:
Here's where we'd integrate the prototype, I think it should be pretty straightforward?

Reply:
Cool!  If I'm reading the code / react goo right, as it is, the Private View exists only in memory (the `privateChanges` state in the definition of `AppView`).  But with Staging Mode, we could update that to drop the second view (the `{suggestedGroceryItemViews}` element under `GroceryListView`) and instead just have the one view that includes the staged changes (which a side-by-side client wouldn't see).  Sound close?

Reply:
Something like that - probably drop SuggestedGroceryItemView as you say and then GroceryItemView would get a prop to indicate that it is a suggested add (so I can draw a green background). Might also need some other tweaks in the bookkeeping so I can remember the deleted items and also which ones are new adds.